views/js/qtiCreator/widgets/interactions/graphicAssociateInteraction/Widget.js

```diff
@@ -13,75 +13,91 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2015 (original work) Open Assessment Technologies SA;
+ * Copyright (c) 2015-2022 (original work) Open Assessment Technologies SA;
  *
  */
 
-
 /**
  * @author Bertrand Chevrier <bertrand@taotesting.com>
  */
 define([
-    'jquery', 'lodash', 'i18n',
+    'jquery',
+    'lodash',
+    'i18n',
     'taoQtiItem/qtiCreator/widgets/interactions/Widget',
     'taoQtiItem/qtiCreator/widgets/interactions/graphicInteraction/Widget',
-    'taoQtiItem/qtiCreator/widgets/interactions/graphicAssociateInteraction/states/states'
-], function($, _, __, Widget, GraphicWidget, states){
-
+    'taoQtiItem/qtiCreator/widgets/interactions/graphicAssociateInteraction/states/states',
+    'taoQtiItem/qtiCommonRenderer/helpers/Graphic'
+], function ($, _, __, Widget, GraphicWidget, states, GraphicHelper) {
     'use strict';
 
     /**
      * The Widget that provides components used by the QTI Creator for the GraphicAssociate Interaction.
-     * 
+     *
      * @extends taoQtiItem/qtiCreator/widgets/interactions/Widget
      * @extends taoQtiItem/qtiCreator/widgets/interactions/GraphicInteraction/Widget
      *
      * @exports taoQtiItem/qtiCreator/widgets/interactions/graphicAssociateInteraction/Widget
-     */      
-    var GraphicAssociateInteractionWidget = _.extend(Widget.clone(), GraphicWidget, {
-
+     */
+    const GraphicAssociateInteractionWidget = _.extend(Widget.clone(), GraphicWidget, {
         /**
          * Set up the widget
          * @see {taoQtiItem/qtiCreator/widgets/interactions/Widget#initCreator}
-         * @param {Object} options - extra options 
+         * @param {Object} options - extra options
          * @param {String} options.baseUrl - the resource base url
          * @param {jQueryElement} options.choiceForm = a reference to the form of the choices
          */
-        initCreator : function(options){
-            var paper;
+        initCreator: function (options) {
             this.baseUrl = options.baseUrl;
             this.choiceForm = options.choiceForm;
-            
+
             this.registerStates(states);
-            
+
             //call parent initCreator
             Widget.initCreator.call(this);
-           
-            paper = this.createPaper(); 
-            if(paper){
+
+            const paper = this.createPaper(_.bind(this.scaleChoices, this));
```

Comment:
**chore:** this should not be any need for using such a trick nowadays...

**suggestion:**
```suggestion
            const paper = this.createPaper(() => this.scaleChoices());
```